pinot-query-planner/src/main/java/org/apache/pinot/query/planner/logical/ShuffleRewriteVisitor.java

```diff
@@ -79,6 +80,16 @@ public Set<Integer> visitWindow(WindowNode node, Void context) {
     throw new UnsupportedOperationException("Window not yet supported!");
   }
 
+  @Override
+  public Set<Integer> visitSetOp(SetOpNode setOpNode, Void context) {
+    StageNode firstInput = setOpNode.getInputs().get(0);
+    if (firstInput instanceof ProjectNode) {
+      Set<Integer> oldPartitionKeys = firstInput.visit(this, context);
+      return deriveNewPartitionKeysFromRexExpressions(((ProjectNode) firstInput).getProjects(), oldPartitionKeys);
+    }
```

Comment:
what's the rational behind this? and was this tested? i don't think this is being used at all.

Reply:
`ShuffleRewriteVisitor` is not used. I think we should remove it.

Reply:
Changed the logic to include all the partition keys from inputs.